pkg/manifests/assets/router/deployment.yaml

```diff
@@ -9,6 +9,7 @@ spec:
     metadata:
       annotations:
         target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+        openshift.io/required-scc: hostnetwork
```

Comment:
The deployment may use "hostnetwork" or "restricted", depending on the ingresscontroller configuration (specifically `IngressController.spec.endpointPublishingStrategy.type`).  So I think we will need `desiredRouterDeployment` in `pkg/operator/controller/ingress/deployment.go` to set this annotation at run-time —right?

By the way, this PR reminds me of some related PRs, #743 and #981, which updated the security context based on previous customer cases where we observed that the wrong SCC could be selected.

Reply:
Thanks for the pointer @Miciah :slightly_smiling_face: I've pushed a change that addresses this.

Reply:
You're right, it seems that we missed this change [from the OCP 4.11 release notes](https://docs.openshift.com/container-platform/4.11/release_notes/ocp-4-11-release-notes.html#ocp-4-11-auth-pod-security-admission):

> The `restricted` SCC is no longer available to users of new clusters, unless the access is explicitly granted. In clusters originally installed in OpenShift Container Platform 4.10 or earlier, all authenticated users can use the `restricted` SCC when upgrading to OpenShift Container Platform 4.11 and later.  

And observations of CI artifacts confirm this:

- periodic-ci-openshift-release-master-ci-4.10-upgrade-from-stable-4.9-e2e-aws-ovn-upgrade has `openshift.io/scc: restricted`.
- periodic-ci-openshift-release-master-ci-4.10-e2e-aws-serial has `openshift.io/scc: restricted`.
- periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-aws-ovn-upgrade has `openshift.io/scc: restricted`.
- periodic-ci-openshift-release-master-ci-4.11-e2e-aws-serial has `openshift.io/scc: hostnetwork`.
- periodic-ci-openshift-release-master-ci-4.12-upgrade-from-stable-4.11-e2e-aws-ovn-upgrade has `openshift.io/scc: hostnetwork`.
- periodic-ci-openshift-release-master-ci-4.12-e2e-aws-sdn-serial has `openshift.io/scc: hostnetwork`.

The intention is definitely that we use "restricted" when we don't need "hostnetwork".  I've filed [OCPBUGS-34373](https://issues.redhat.com/browse/OCPBUGS-34373) and posted #1064 to address this oversight.  Thanks for pointing it out!

Reply:
Glad this helped :) I'll wait for #1064 to merge and then rebase this and try again :+1: